File: 3191_binary_array.cpp
#include <iostream>

using namespace std;

int solve(std::vector<int> nums) {
    int ans = 0;

    for (int i=0; i<nums.size(); i++) {
        if (nums[i] == 0) {
            if (i+2 >= nums.size()) return -1;

            ans += 1;
            nums[i] ^= 1;
            nums[i + 1] ^= 1;
            nums[i + 2] ^= 1;
        }
    }

    return ans;
}

int main() {
    cout << solve({0,1,1,1,0,0}) << endl;
    cout << solve({0,1,1,1}) << endl;
}

File: 3191_binary_array.py
def solve(nums):
    ans = 0
    for i in range(len(nums)):
        if nums[i] == 0:
            if i+2 >= len(nums):
                return -1
            nums[i] = (nums[i] + 1) % 2
            nums[i + 1] = (nums[i + 1] + 1) % 2
            nums[i + 2] = (nums[i + 2] + 1) % 2
            ans += 1
    return ans


print(solve([0, 1, 1, 1, 0, 0]))
print(solve([0, 1, 1, 1]))
